cognite_toolkit/_cdf_tk/loaders/_resource_loaders/function_loaders.py

```diff
@@ -533,22 +523,33 @@ def retrieve(self, ids: SequenceNotStr[FunctionScheduleID]) -> FunctionSchedules
 
     def create(self, items: FunctionScheduleWriteList) -> FunctionSchedulesList:
         created_list = FunctionSchedulesList([], cognite_client=self.client)
+        functions_to_lookup = list({item.function_external_id for item in items if item.function_external_id})
+        function_id_by_external_id: dict[str, int] = {}
+        if functions_to_lookup:
+            function_ids = self.client.lookup.functions.id(functions_to_lookup)
+            function_id_by_external_id = dict(zip(functions_to_lookup, function_ids))
+
         for item in items:
             id_ = self.get_id(item)
+            if id_.function_external_id not in function_id_by_external_id:
+                raise ToolkitRequiredValueError(f"Function {id_.function_external_id!r} not found for schedule {id_!r}")
             if id_ not in self.authentication_by_id:
                 raise ToolkitRequiredValueError(f"Authentication is missing for schedule {id_!r}")
             client_credentials = self.authentication_by_id[id_]
             try:
-                created = self.client.functions.schedules.create(item, client_credentials=client_credentials)
+                session = self.client.iam.sessions.create(client_credentials, session_type="CLIENT_CREDENTIALS")
```

Comment:
Make this call explicit, instead of indirectly in the `client.functions.schedules.create`, makes it easier to trace that it was the credentials failing.